src/GenAPI/Microsoft.DotNet.GenAPI.Task/build/Microsoft.DotNet.GenAPI.Task.targets

```
@@ -0,0 +1,60 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Project>
+  <PropertyGroup>
+    <DotNetGenAPITaskAssembly Condition="'$(MSBuildRuntimeType)' == 'Core' and Exists('$(MSBuildThisFileDirectory)..\tools\net7.0\Microsoft.DotNet.GenAPI.Task.dll')">$(MSBuildThisFileDirectory)..\tools\net7.0\Microsoft.DotNet.GenAPI.Task.dll</DotNetGenAPITaskAssembly>
+    <DotNetGenAPITaskAssembly Condition="'$(MSBuildRuntimeType)' == 'Core' and '$(DotNetGenAPITaskAssembly)' == ''">$(MSBuildThisFileDirectory)..\tools\net6.0\Microsoft.DotNet.GenAPI.Task.dll</DotNetGenAPITaskAssembly>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- Hook onto the TargetsTriggeredByCompilation target which only runs when the compiler is invoked. -->
+    <TargetsTriggeredByCompilation Condition="'$(GenAPIGenerateReferenceAssemblySource)' == 'true'">
+      $(TargetsTriggeredByCompilation);
+      GenAPIGenerateReferenceAssemblySource
+    </TargetsTriggeredByCompilation>
+  </PropertyGroup>
+
+  <!-- Only run this target when the GenAPIGenerateReferenceAssemblySource target is invoked directly
+       which means that GenAPIGenerateReferenceAssemblySource property is false. -->
+  <Target Name="_GenAPIDontBuildProjectReferences" 
+          Condition="'$(GenAPIGenerateReferenceAssemblySource)' != 'true'">
+    <PropertyGroup>
+      <BuildProjectReferences>false</BuildProjectReferences>
+    </PropertyGroup>
+  </Target>
+
+  <!-- GenAPIGenerateReferenceSource was the old target's name. Keep it to not break existing dev workflows. -->
+  <Target Name="GenAPIGenerateReferenceSource"
+          DependsOnTargets="GenAPIGenerateReferenceAssemblySource" />
+
+  <UsingTask TaskName="Microsoft.DotNet.GenAPI.Task.GenAPITask" AssemblyFile="$(DotNetGenAPITaskAssembly)" />
+
+  <Target Name="GenAPIGenerateReferenceAssemblySource"
+          DependsOnTargets="_GenAPIDontBuildProjectReferences;ResolveAssemblyReferences">
```

Use the `FindReferenceAssembliesForReferences` target instead, which calculates the actual reference assemblies.

```suggestion
          DependsOnTargets="_GenAPIDontBuildProjectReferences;FindReferenceAssembliesForReferences">
```

---

Updated.